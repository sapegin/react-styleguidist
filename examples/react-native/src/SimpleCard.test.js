import renderer from 'react-test-renderer';

import React from 'react';
import { Text } from 'react-native';
import SimpleCard from './SimpleCard';

it('renders without crashing', () => {
	const rendered = renderer
		.create(
			<SimpleCard title="Basic Card">
				<Text>Basic example 1</Text>
			</SimpleCard>
		)
		.toJSON();
	expect(rendered).toBeTruthy();
});

it('Can change card color', () => {
	const rendered = renderer.create(
		<SimpleCard title="Basic Card" color="red">
			<Text>Basic example 1</Text>
		</SimpleCard>
	);
	expect(rendered).toMatchSnapshot();
});

it('Can change title text color', () => {
	const rendered = renderer.create(
		<SimpleCard title="Basic Card" titleTextColor="red">
			<Text>Basic example 1</Text>
		</SimpleCard>
	);
	expect(rendered).toMatchSnapshot();
});
