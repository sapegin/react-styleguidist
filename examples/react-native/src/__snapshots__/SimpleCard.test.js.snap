// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can change card color 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderWidth": 1,
        "marginBottom": 12,
      },
      Object {
        "borderColor": "red",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "padding": 12,
        },
        Object {
          "backgroundColor": "red",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "600",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Basic Card
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 12,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Basic example 1
    </Text>
  </View>
</View>
`;

exports[`Can change title text color 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderWidth": 1,
        "marginBottom": 12,
      },
      Object {
        "borderColor": "#333",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "padding": 12,
        },
        Object {
          "backgroundColor": "#333",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "600",
          },
          Object {
            "color": "red",
          },
        ]
      }
    >
      Basic Card
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 12,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Basic example 1
    </Text>
  </View>
</View>
`;
